### DISCUSSION TITLE
Openai API broken with llama.cpp

### DISCUSSION THREAD

**Discussion by inversetrs** [open-webui/open-webui#5856] [General]
# Bug Report

## Installation Method

[Describe the method you used to install the project, e.g., git clone, Docker, pip, etc.]

## Environment

- **Open WebUI Version: git master

- **Operating System: Linux

**Confirmation:**

- [x ] I have read and followed all the instructions provided in the README.md.
- [ x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ x] I have included the Docker container logs.
- [ x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

Get available model in the interface

## Actual Behavior:

Connects to openapi llama.cpp server, gets correct response, but the interface show no models available.

## Description

**Bug Summary:**
It used to work a few weeks ago, something got lost in the latest commits.

## Reproduction Details

**Steps to Reproduce:**
Starts lamma.cpp server, add data in the admin panel, connection sucessfull, get_all_models does not return an error, however, no model shows up in the interface. 

## Additional Information
localhost/v1/models returns the loaded model correctly.

**Comment by tjbck** [comment]
Cannot reproduce, let us know your findings!

**Comment by inversetrs** [comment]
Just tried a fresh install using pip. It crashes because there is no ollama running. Setting the enviroment variable ENABLE_OLLAMA_API=false allows the server to start.  In the admin panel it confirms "Server connection verified
" to the local llama.cpp server, that confirms the request: GET /v1/models 127.0.0.1 , however the available models list remains empty. It was working a few releases prior. Let me know if I can help.

**Comment by strawberrymelonpanda** [comment]
Hey, was having similar issues and ran into this thread.  It eventually worked for me after:
1) Started the server with `ENABLE_OLLAMA_API=false` as @inversetrs suggested to get it to start.
(`ENABLE_OLLAMA_API=false open-webui serve --host 127.0.0.1 --port 5001`)
2) Clicked my username (lower left) > settings > admin settings > connections
3) Made sure OpenAI API is toggled on and changed OpenAI's URL to http://127.0.0.1:8080/v1

Then it detected the model I started Llama.CPP with and everything seems fine.
(`./llama-server --model "<gguf path>" --ctx-size 32000 --gpu-layers 99`)

_Ubuntu 22.04.4 LTS on WSL2
Llama.cpp server build: 3855 (a90484c6)
Open WebUI v0.3.32_
